code/datums/components/ntnet_interface.dm

```diff
@@ -1,66 +1,136 @@
-//Thing meant for allowing datums and objects to access an NTnet network datum.
 /datum/proc/ntnet_receive(datum/netdata/data)
-	return
+	SEND_SIGNAL(src, COMSIG_COMPONENT_NTNET_RECEIVE, data)
 
-/datum/proc/ntnet_receive_broadcast(datum/netdata/data)
-	return
 
-/datum/proc/ntnet_send(datum/netdata/data, netid)
+// helper function.  So you don't have to get the component
+/datum/proc/ntnet_send(datum/netdata/data)
 	var/datum/component/ntnet_interface/NIC = GetComponent(/datum/component/ntnet_interface)
 	if(!NIC)
 		return FALSE
-	return NIC.__network_send(data, netid)
+	data.sender_id = NIC.hardware_id
+	data.network_id = NIC.network.network_id
+	return SSnetworks.transmit(data)
+
 
 /datum/component/ntnet_interface
-	var/hardware_id			//text. this is the true ID. do not change this. stuff like ID forgery can be done manually.
-	var/network_name = ""			//text
-	var/list/networks_connected_by_id = list()		//id = datum/ntnet
-	var/differentiate_broadcast = TRUE				//If false, broadcasts go to ntnet_receive. NOT RECOMMENDED.
+	var/hardware_id = null				// text. this is the true ID. do not change this. stuff like ID forgery can be done manually.
```

Comment:
The documentation for these class-level vars does not meet our [code documentation standards](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md), please add or improve the documentation here.